src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -31,77 +31,115 @@ namespace Microsoft.Azure.Commands.Insights.Diagnostics
     [Cmdlet(VerbsCommon.Set, "AzureRmDiagnosticSetting", SupportsShouldProcess = true), OutputType(typeof(PSServiceDiagnosticSettings))]
     public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
     {
+        internal const string SetAzureRmDiagnosticSettingOldParamGroup = "OldSetDiagnosticSetting";
+        internal const string SetAzureRmDiagnosticSettingNewParamGroup = "NewSetDiagnosticSetting";
+
         public const string StorageAccountIdParamName = "StorageAccountId";
         public const string ServiceBusRuleIdParamName = "ServiceBusRuleId";
+        public const string EventHubNameParamName = "EventHubName";
         public const string EventHubRuleIdParamName = "EventHubAuthorizationRuleId";
         public const string WorkspacetIdParamName = "WorkspaceId";
         public const string EnabledParamName = "Enabled";
 
+        /// <summary>
+        /// This is a temporary constant to provide backwards compatibility
+        /// </summary>
+        internal const string TempServiceName = "service";
+
         #region Parameters declarations
 
+        /// <summary>
+        /// Gets or sets the InputObject parameter of the cmdlet
+        /// </summary>
+        [Parameter(ParameterSetName = SetAzureRmDiagnosticSettingNewParamGroup, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The complete spec of a DiagnosticSettingSetting")]
```

Comment:
@pkamath13 @gucalder change `ValueFromPipelineByPropertyName` to `ValueFromPipeline`

Reply:
@cormacpayne  is this valid for all parameters and all cmdlets?  Including the parameter to implement the pipeline feature?

Reply:
@pkamath13 @gucalder any `-InputObject` parameter should have `ValueFromPipeline = true`

Reply:
@gucalder the other parameters can remain as they are